docs/release-notes/v0.7.0.rst

```diff
@@ -0,0 +1,254 @@
+|release v0.7.0|_
+=================
+
+This is a minor release from ``v0.6.3`` → ``v0.7.0``.
+
+Important Notes
+---------------
+
+* Please note this release has **API breaking changes** and carefully read these
```

Comment:
For completeness, I have some suggestions based on the changes required in `cabinetry` ( https://github.com/scikit-hep/cabinetry/pull/301) that developers may be interested in:
- `paramset.suggested_fixed` has changed behavior (old behavior is achieved via `paramset.suggested_fixed_as_bool`, see #1639),
- #1625 changed the order of model parameters.

While a different order of parameters is not necessarily a breaking change, I think it would be useful to highlight this explicitly as it could lead to confusion when comparing e.g. fit results with the old / new version without being aware that a suitable reordering is required.

Reply:
The additions look good to me, thanks!

Reply:
Thanks for your super helpful feedback! :pray: